### ISSUE TITLE
DCA tl_form_field "toggleIcon" method not working as expected

### ISSUE THREAD

**Issue by cmartin6** [contao/contao#3851]
### Affected version(s)

4.9.21-4.9.24

### Description

If I press the "toggleIcon" button of a form-element and reload the page, the item is not set to "invisible". I managed to find the according error-message in the logs: `Argument 1 passed to Contao\DcaLoader::{closure}() must be an instance of Contao\DataContainer, null given`

I debugged into the code of the dca file `tl_form_field.php` and noticed that` func_num_args()` returns 6 on line 778 - why does the method call does not feature the `Contao/DataContainer` instance as argument 12 as expected?

following is my composer.json:
`{
    "type": "project",
    "require": {
        "bwein-net/contao-backend-customizer": "^1.1",
        "codefog/contao-news_categories": "^3.1",
        "contao/conflicts": "*@dev",
        "contao/listing-bundle": "4.9.*",
        "contao/manager-bundle": "4.9.*",
        "contao/news-bundle": "4.9.*",
        "delahaye/dlh_googlemaps": "^2.4",
        "dma/dma_elementgenerator": "dev-managed-edition",
        "duncrow-gmbh/polymode": "^1.1",
        "felixpfeiffer/subcolumns": "^3.1",
        "hofff/contao-calendarfield": "^3.0",
        "ma3xl3/contao-easy-favicon": "^2.0",
        "netzmacht/contao-font-awesome-inserttag": "^2.2",
        "numero2/contao-opengraph3": "^4.2",
        "oveleon/contao-cookiebar": "^1.9",
        "psi/boxes4ward": "^2.0",
        "richardhj/contao-backup-manager": "^2.2",
        "terminal42/contao-folderpage": "^2.0",
        "terminal42/contao-leads": "^1.4",
        "terminal42/notification_center": "^1.5",
        "heimrichhannot/contao-slick-bundle": "^1.6",
        "terminal42/contao-changelanguage": "^3.2",
        "madeyourday/contao-rocksolid-custom-elements": "^2.3",
        "inspiredminds/contao-wowjs": "^1.1"
    },
    "extra": {
        "contao-component-dir": "assets"
    },
    "autoload": {
        "psr-4": {
            "": "src/"
        }
    },
    "scripts": {
        "post-install-cmd": [
            "Contao\\ManagerBundle\\Composer\\ScriptHandler::initializeApplication"
        ],
        "post-update-cmd": [
            "Contao\\ManagerBundle\\Composer\\ScriptHandler::initializeApplication"
        ]
    },
    "require-dev": {
        "deployer/deployer": "^6.8",
        "deployer/recipes": "^6.2",
        "terminal42/deployer-recipes": "dev-main@dev"
    }
}
`

**Comment by fritzmg** [comment]
> If I press the "toggleIcon" button of a form-element and reload the page, the item is not set to "invisible". I managed to find the according error-message in the logs: `Argument 1 passed to Contao\DcaLoader::{closure}() must be an instance of Contao\DataContainer, null given`

Use the debug mode and post the complete stack trace please.

**Comment by cmartin6** [comment]
How do I use the debug mode in combination with ajax-requests from the backend?

**Comment by fritzmg** [comment]
Look at the Symfony toolbar - there 🔴 you will find the AJAX requests and you can open the profile of each request 🟡:

![image](https://user-images.githubusercontent.com/4970961/146811206-68fafccc-1778-4e92-80af-75f0bfa45100.png)

**Comment by cmartin6** [comment]
Thank you Fritz - here is the complete stack trace of the specific AJAX request: 

`TypeError:
Argument 1 passed to Contao\DcaLoader::{closure}() must be an instance of Contao\DataContainer, null given, called in /Users/clemensmartin/Sites/traubebraz/var/cache/dev/contao/dca/tl_form_field.php on line 835

  at var/cache/dev/contao/dca/tl_form_field.php:1144
  at Contao\DcaLoader->{closure}(null)
     (var/cache/dev/contao/dca/tl_form_field.php:835)
  at tl_form_field->toggleVisibility('47', false, null)
     (var/cache/dev/contao/dca/tl_form_field.php:779)
  at tl_form_field->toggleIcon(array('id' => '316', 'pid' => '5', 'sorting' => '128', 'tstamp' => '1640018151', 'invisible' => '', 'type' => 'explanation', 'name' => '', 'label' => '', 'text' => '<h3>Ihre Reisedaten</h3><p>Mit einem * markierte Felder sind Pflichtfelder.</p><hr>', 'html' => null, 'options' => null, 'mandatory' => '', 'rgxp' => '', 'placeholder' => '', 'minlength' => '0', 'maxlength' => '0', 'size' => 'a:2:{i:0;i:4;i:1;i:40;}', 'multiple' => '', 'mSize' => '0', 'extensions' => 'jpg,jpeg,gif,png,pdf,doc,xls,ppt', 'storeFile' => '', 'uploadFolder' => null, 'useHomeDir' => '', 'doNotOverwrite' => '', 'class' => '', 'value' => '', 'accesskey' => '', 'tabindex' => '0', 'fSize' => '0', 'customTpl' => '', 'slabel' => '', 'imageSubmit' => '', 'singleSRC' => null, 'fsc_type' => '', 'fsc_name' => '', 'fsc_gapuse' => '', 'fsc_gap' => '0', 'fsc_equalize' => '', 'fsc_color' => '', 'fsc_parent' => '0', 'fsc_childs' => '', 'fsc_sortid' => '0', 'dateFormat' => '', 'dateDirection' => '', 'dateParseValue' => '', 'dateIncludeCSS' => '', 'dateIncludeCSSTheme' => 'smoothness', 'dateImage' => '1', 'dateImageSRC' => null, 'dateDisabledWeekdays' => null, 'dateDisabledDays' => '', 'leadStore' => '', 'step' => '', 'maxval' => '', 'minval' => '', 'autogrid_id' => '0', 'autogrid_class' => '', 'autogrid_sameheight' => '', 'autogrid_gutter' => '', 'autogrid_align_mobile' => '', 'autogrid_align' => '', 'autogrid_tablet' => '', 'autogrid_mobile' => '', 'autogrid_css' => '', 'margin_b_m' => '', 'margin_t_m' => '', 'margin_b' => '', 'margin_t' => '', 'autogrid' => '', 'autogrid_offset' => '', 'rsce_data' => null, 'wowjsIteration' => null, 'wowjsOffset' => null, 'wowjsDelay' => '', 'wowjsDuration' => '', 'wowjsAnimation' => ''), null, 'Formularfeld ID 316 verstecken/anzeigen', 'Formularfeld ID 316 verstecken/anzeigen', 'visible.svg', ' class="toggle" onclick="Backend.getScrollOffset();return AjaxRequest.toggleVisibility(this,316)"')
     (var/cache/dev/contao/dca/tl_form_field.php:1667)
  at tl_form_subcols->toggleIcons(array('id' => '316', 'pid' => '5', 'sorting' => '128', 'tstamp' => '1640018151', 'invisible' => '', 'type' => 'explanation', 'name' => '', 'label' => '', 'text' => '<h3>Ihre Reisedaten</h3><p>Mit einem * markierte Felder sind Pflichtfelder.</p><hr>', 'html' => null, 'options' => null, 'mandatory' => '', 'rgxp' => '', 'placeholder' => '', 'minlength' => '0', 'maxlength' => '0', 'size' => 'a:2:{i:0;i:4;i:1;i:40;}', 'multiple' => '', 'mSize' => '0', 'extensions' => 'jpg,jpeg,gif,png,pdf,doc,xls,ppt', 'storeFile' => '', 'uploadFolder' => null, 'useHomeDir' => '', 'doNotOverwrite' => '', 'class' => '', 'value' => '', 'accesskey' => '', 'tabindex' => '0', 'fSize' => '0', 'customTpl' => '', 'slabel' => '', 'imageSubmit' => '', 'singleSRC' => null, 'fsc_type' => '', 'fsc_name' => '', 'fsc_gapuse' => '', 'fsc_gap' => '0', 'fsc_equalize' => '', 'fsc_color' => '', 'fsc_parent' => '0', 'fsc_childs' => '', 'fsc_sortid' => '0', 'dateFormat' => '', 'dateDirection' => '', 'dateParseValue' => '', 'dateIncludeCSS' => '', 'dateIncludeCSSTheme' => 'smoothness', 'dateImage' => '1', 'dateImageSRC' => null, 'dateDisabledWeekdays' => null, 'dateDisabledDays' => '', 'leadStore' => '', 'step' => '', 'maxval' => '', 'minval' => '', 'autogrid_id' => '0', 'autogrid_class' => '', 'autogrid_sameheight' => '', 'autogrid_gutter' => '', 'autogrid_align_mobile' => '', 'autogrid_align' => '', 'autogrid_tablet' => '', 'autogrid_mobile' => '', 'autogrid_css' => '', 'margin_b_m' => '', 'margin_t_m' => '', 'margin_b' => '', 'margin_t' => '', 'autogrid' => '', 'autogrid_offset' => '', 'rsce_data' => null, 'wowjsIteration' => null, 'wowjsOffset' => null, 'wowjsDelay' => '', 'wowjsDuration' => '', 'wowjsAnimation' => ''), null, 'Formularfeld ID 316 verstecken/anzeigen', 'Formularfeld ID 316 verstecken/anzeigen', 'visible.svg', ' class="toggle" onclick="Backend.getScrollOffset();return AjaxRequest.toggleVisibility(this,316)"', 'tl_form_field', null, null, false, null, '45', object(DC_Table))
     (vendor/contao/core-bundle/src/Resources/contao/classes/DataContainer.php:802)
  at Contao\DataContainer->generateButtons(array('id' => '316', 'pid' => '5', 'sorting' => '128', 'tstamp' => '1640018151', 'invisible' => '', 'type' => 'explanation', 'name' => '', 'label' => '', 'text' => '<h3>Ihre Reisedaten</h3><p>Mit einem * markierte Felder sind Pflichtfelder.</p><hr>', 'html' => null, 'options' => null, 'mandatory' => '', 'rgxp' => '', 'placeholder' => '', 'minlength' => '0', 'maxlength' => '0', 'size' => 'a:2:{i:0;i:4;i:1;i:40;}', 'multiple' => '', 'mSize' => '0', 'extensions' => 'jpg,jpeg,gif,png,pdf,doc,xls,ppt', 'storeFile' => '', 'uploadFolder' => null, 'useHomeDir' => '', 'doNotOverwrite' => '', 'class' => '', 'value' => '', 'accesskey' => '', 'tabindex' => '0', 'fSize' => '0', 'customTpl' => '', 'slabel' => '', 'imageSubmit' => '', 'singleSRC' => null, 'fsc_type' => '', 'fsc_name' => '', 'fsc_gapuse' => '', 'fsc_gap' => '0', 'fsc_equalize' => '', 'fsc_color' => '', 'fsc_parent' => '0', 'fsc_childs' => '', 'fsc_sortid' => '0', 'dateFormat' => '', 'dateDirection' => '', 'dateParseValue' => '', 'dateIncludeCSS' => '', 'dateIncludeCSSTheme' => 'smoothness', 'dateImage' => '1', 'dateImageSRC' => null, 'dateDisabledWeekdays' => null, 'dateDisabledDays' => '', 'leadStore' => '', 'step' => '', 'maxval' => '', 'minval' => '', 'autogrid_id' => '0', 'autogrid_class' => '', 'autogrid_sameheight' => '', 'autogrid_gutter' => '', 'autogrid_align_mobile' => '', 'autogrid_align' => '', 'autogrid_tablet' => '', 'autogrid_mobile' => '', 'autogrid_css' => '', 'margin_b_m' => '', 'margin_t_m' => '', 'margin_b' => '', 'margin_t' => '', 'autogrid' => '', 'autogrid_offset' => '', 'rsce_data' => null, 'wowjsIteration' => null, 'wowjsOffset' => null, 'wowjsDelay' => '', 'wowjsDuration' => '', 'wowjsAnimation' => ''), 'tl_form_field', null, false, null, null, '45')
     (vendor/contao/core-bundle/src/Resources/contao/drivers/DC_Table.php:4512)
  at Contao\DC_Table->parentView()
     (vendor/contao/core-bundle/src/Resources/contao/drivers/DC_Table.php:350)
  at Contao\DC_Table->showAll()
     (vendor/contao/core-bundle/src/Resources/contao/classes/Backend.php:644)
  at Contao\Backend->getBackendModule('form', null)
     (vendor/contao/core-bundle/src/Resources/contao/controllers/BackendMain.php:162)
  at Contao\BackendMain->run()
     (vendor/contao/core-bundle/src/Controller/BackendController.php:48)
  at Contao\CoreBundle\Controller\BackendController->mainAction()
     (vendor/symfony/http-kernel/HttpKernel.php:158)
  at Symfony\Component\HttpKernel\HttpKernel->handleRaw(object(Request), 1)
     (vendor/symfony/http-kernel/HttpKernel.php:80)
  at Symfony\Component\HttpKernel\HttpKernel->handle(object(Request), 1, true)
     (vendor/symfony/http-kernel/Kernel.php:201)
  at Symfony\Component\HttpKernel\Kernel->handle(object(Request))
     (web/index.php:31)
  at require('/Users/clemensmartin/Sites/traubebraz/web/index.php')
     (web/app.php:4)                `

**Comment by fritzmg** [comment]
Please use three backticks to post the stack. Also your Contao internal cache seems to be built, even though you are in dev mode. Please delete `var/cache/dev` and try again (do not run `cache:warmup --env=dev`).

**Comment by cmartin6** [comment]
I deleted the `var/cache/dev` directory - here is the stack trace:

```
TypeError:
Argument 1 passed to Contao\DcaLoader::{closure}() must be an instance of Contao\DataContainer, null given, called in /Users/clemensmartin/Sites/traubebraz/vendor/contao/core-bundle/src/Resources/contao/dca/tl_form_field.php on line 834

  at vendor/inspiredminds/contao-wowjs/src/Resources/contao/dca/tl_form_field.php:64
  at Contao\DcaLoader->{closure}(null)
     (vendor/contao/core-bundle/src/Resources/contao/dca/tl_form_field.php:834)
  at tl_form_field->toggleVisibility('47', false, null)
     (vendor/contao/core-bundle/src/Resources/contao/dca/tl_form_field.php:778)
  at tl_form_field->toggleIcon(array('id' => '316', 'pid' => '5', 'sorting' => '128', 'tstamp' => '1640018151', 'invisible' => '', 'type' => 'explanation', 'name' => '', 'label' => '', 'text' => '<h3>Ihre Reisedaten</h3><p>Mit einem * markierte Felder sind Pflichtfelder.</p><hr>', 'html' => null, 'options' => null, 'mandatory' => '', 'rgxp' => '', 'placeholder' => '', 'minlength' => '0', 'maxlength' => '0', 'size' => 'a:2:{i:0;i:4;i:1;i:40;}', 'multiple' => '', 'mSize' => '0', 'extensions' => 'jpg,jpeg,gif,png,pdf,doc,xls,ppt', 'storeFile' => '', 'uploadFolder' => null, 'useHomeDir' => '', 'doNotOverwrite' => '', 'class' => '', 'value' => '', 'accesskey' => '', 'tabindex' => '0', 'fSize' => '0', 'customTpl' => '', 'slabel' => '', 'imageSubmit' => '', 'singleSRC' => null, 'fsc_type' => '', 'fsc_name' => '', 'fsc_gapuse' => '', 'fsc_gap' => '0', 'fsc_equalize' => '', 'fsc_color' => '', 'fsc_parent' => '0', 'fsc_childs' => '', 'fsc_sortid' => '0', 'dateFormat' => '', 'dateDirection' => '', 'dateParseValue' => '', 'dateIncludeCSS' => '', 'dateIncludeCSSTheme' => 'smoothness', 'dateImage' => '1', 'dateImageSRC' => null, 'dateDisabledWeekdays' => null, 'dateDisabledDays' => '', 'leadStore' => '', 'step' => '', 'maxval' => '', 'minval' => '', 'autogrid_id' => '0', 'autogrid_class' => '', 'autogrid_sameheight' => '', 'autogrid_gutter' => '', 'autogrid_align_mobile' => '', 'autogrid_align' => '', 'autogrid_tablet' => '', 'autogrid_mobile' => '', 'autogrid_css' => '', 'margin_b_m' => '', 'margin_t_m' => '', 'margin_b' => '', 'margin_t' => '', 'autogrid' => '', 'autogrid_offset' => '', 'rsce_data' => null, 'wowjsIteration' => null, 'wowjsOffset' => null, 'wowjsDelay' => '', 'wowjsDuration' => '', 'wowjsAnimation' => ''), null, 'Formularfeld ID 316 verstecken/anzeigen', 'Formularfeld ID 316 verstecken/anzeigen', 'visible.svg', ' class="toggle" onclick="Backend.getScrollOffset();return AjaxRequest.toggleVisibility(this,316)"')
     (vendor/felixpfeiffer/subcolumns/dca/tl_form_field.php:508)
  at tl_form_subcols->toggleIcons(array('id' => '316', 'pid' => '5', 'sorting' => '128', 'tstamp' => '1640018151', 'invisible' => '', 'type' => 'explanation', 'name' => '', 'label' => '', 'text' => '<h3>Ihre Reisedaten</h3><p>Mit einem * markierte Felder sind Pflichtfelder.</p><hr>', 'html' => null, 'options' => null, 'mandatory' => '', 'rgxp' => '', 'placeholder' => '', 'minlength' => '0', 'maxlength' => '0', 'size' => 'a:2:{i:0;i:4;i:1;i:40;}', 'multiple' => '', 'mSize' => '0', 'extensions' => 'jpg,jpeg,gif,png,pdf,doc,xls,ppt', 'storeFile' => '', 'uploadFolder' => null, 'useHomeDir' => '', 'doNotOverwrite' => '', 'class' => '', 'value' => '', 'accesskey' => '', 'tabindex' => '0', 'fSize' => '0', 'customTpl' => '', 'slabel' => '', 'imageSubmit' => '', 'singleSRC' => null, 'fsc_type' => '', 'fsc_name' => '', 'fsc_gapuse' => '', 'fsc_gap' => '0', 'fsc_equalize' => '', 'fsc_color' => '', 'fsc_parent' => '0', 'fsc_childs' => '', 'fsc_sortid' => '0', 'dateFormat' => '', 'dateDirection' => '', 'dateParseValue' => '', 'dateIncludeCSS' => '', 'dateIncludeCSSTheme' => 'smoothness', 'dateImage' => '1', 'dateImageSRC' => null, 'dateDisabledWeekdays' => null, 'dateDisabledDays' => '', 'leadStore' => '', 'step' => '', 'maxval' => '', 'minval' => '', 'autogrid_id' => '0', 'autogrid_class' => '', 'autogrid_sameheight' => '', 'autogrid_gutter' => '', 'autogrid_align_mobile' => '', 'autogrid_align' => '', 'autogrid_tablet' => '', 'autogrid_mobile' => '', 'autogrid_css' => '', 'margin_b_m' => '', 'margin_t_m' => '', 'margin_b' => '', 'margin_t' => '', 'autogrid' => '', 'autogrid_offset' => '', 'rsce_data' => null, 'wowjsIteration' => null, 'wowjsOffset' => null, 'wowjsDelay' => '', 'wowjsDuration' => '', 'wowjsAnimation' => ''), null, 'Formularfeld ID 316 verstecken/anzeigen', 'Formularfeld ID 316 verstecken/anzeigen', 'visible.svg', ' class="toggle" onclick="Backend.getScrollOffset();return AjaxRequest.toggleVisibility(this,316)"', 'tl_form_field', null, null, false, null, '45', object(DC_Table))
     (vendor/contao/core-bundle/src/Resources/contao/classes/DataContainer.php:802)
  at Contao\DataContainer->generateButtons(array('id' => '316', 'pid' => '5', 'sorting' => '128', 'tstamp' => '1640018151', 'invisible' => '', 'type' => 'explanation', 'name' => '', 'label' => '', 'text' => '<h3>Ihre Reisedaten</h3><p>Mit einem * markierte Felder sind Pflichtfelder.</p><hr>', 'html' => null, 'options' => null, 'mandatory' => '', 'rgxp' => '', 'placeholder' => '', 'minlength' => '0', 'maxlength' => '0', 'size' => 'a:2:{i:0;i:4;i:1;i:40;}', 'multiple' => '', 'mSize' => '0', 'extensions' => 'jpg,jpeg,gif,png,pdf,doc,xls,ppt', 'storeFile' => '', 'uploadFolder' => null, 'useHomeDir' => '', 'doNotOverwrite' => '', 'class' => '', 'value' => '', 'accesskey' => '', 'tabindex' => '0', 'fSize' => '0', 'customTpl' => '', 'slabel' => '', 'imageSubmit' => '', 'singleSRC' => null, 'fsc_type' => '', 'fsc_name' => '', 'fsc_gapuse' => '', 'fsc_gap' => '0', 'fsc_equalize' => '', 'fsc_color' => '', 'fsc_parent' => '0', 'fsc_childs' => '', 'fsc_sortid' => '0', 'dateFormat' => '', 'dateDirection' => '', 'dateParseValue' => '', 'dateIncludeCSS' => '', 'dateIncludeCSSTheme' => 'smoothness', 'dateImage' => '1', 'dateImageSRC' => null, 'dateDisabledWeekdays' => null, 'dateDisabledDays' => '', 'leadStore' => '', 'step' => '', 'maxval' => '', 'minval' => '', 'autogrid_id' => '0', 'autogrid_class' => '', 'autogrid_sameheight' => '', 'autogrid_gutter' => '', 'autogrid_align_mobile' => '', 'autogrid_align' => '', 'autogrid_tablet' => '', 'autogrid_mobile' => '', 'autogrid_css' => '', 'margin_b_m' => '', 'margin_t_m' => '', 'margin_b' => '', 'margin_t' => '', 'autogrid' => '', 'autogrid_offset' => '', 'rsce_data' => null, 'wowjsIteration' => null, 'wowjsOffset' => null, 'wowjsDelay' => '', 'wowjsDuration' => '', 'wowjsAnimation' => ''), 'tl_form_field', null, false, null, null, '45')
     (vendor/contao/core-bundle/src/Resources/contao/drivers/DC_Table.php:4512)
  at Contao\DC_Table->parentView()
     (vendor/contao/core-bundle/src/Resources/contao/drivers/DC_Table.php:350)
  at Contao\DC_Table->showAll()
     (vendor/contao/core-bundle/src/Resources/contao/classes/Backend.php:644)
  at Contao\Backend->getBackendModule('form', null)
     (vendor/contao/core-bundle/src/Resources/contao/controllers/BackendMain.php:162)
  at Contao\BackendMain->run()
     (vendor/contao/core-bundle/src/Controller/BackendController.php:48)
  at Contao\CoreBundle\Controller\BackendController->mainAction()
     (vendor/symfony/http-kernel/HttpKernel.php:158)
  at Symfony\Component\HttpKernel\HttpKernel->handleRaw(object(Request), 1)
     (vendor/symfony/http-kernel/HttpKernel.php:80)
  at Symfony\Component\HttpKernel\HttpKernel->handle(object(Request), 1, true)
     (vendor/symfony/http-kernel/Kernel.php:201)
  at Symfony\Component\HttpKernel\Kernel->handle(object(Request))
     (web/index.php:31)
  at require('/Users/clemensmartin/Sites/traubebraz/web/index.php')
     (web/app.php:4)

```

**Comment by fritzmg** [comment]
This is an issue with [inspiredminds/contao-wowjs](https://github.com/inspiredminds/contao-wowjs) and not an issue with Contao. It is now fixed in version 1.1.3 of the extension.
